src/test/java/net/snowflake/client/jdbc/DatabaseMetaDataLatestIT.java

```diff
@@ -1694,4 +1694,172 @@ public void testUpdateLocatorsCopyUnsupported() throws SQLException {
+    // Should return result for matching schema and table name
+    assertEquals(1, getSizeOfResultSet(dbmd.getExportedKeys(database, schema, table1)));
+
+    // Should return an empty result if we try a pattern match on the schema
+    assertEquals(0, getSizeOfResultSet(dbmd.getExportedKeys(database, schemaPattern, table1)));
+
+    // Should return an empty result if we try a pattern match on the table name
+    assertEquals(0, getSizeOfResultSet(dbmd.getExportedKeys(database, null, "T%")));
+
+    // Should return result for matching schema and table name
+    assertEquals(
+        1,
+        getSizeOfResultSet(
+            dbmd.getCrossReference(database, schema, table1, database, schema, table2)));
+
+    // Should return an empty result if we try a pattern match on any of the table or schema names
+    assertEquals(
+        0,
+        getSizeOfResultSet(
+            dbmd.getCrossReference(database, schemaPattern, table1, database, schema, table2)));
+    assertEquals(
+        0,
+        getSizeOfResultSet(
+            dbmd.getCrossReference(database, schema, table1, database, schemaPattern, table2)));
+    assertEquals(
+        0,
+        getSizeOfResultSet(dbmd.getCrossReference(database, null, "T%", database, schema, table2)));
+    assertEquals(
+        0,
+        getSizeOfResultSet(dbmd.getCrossReference(database, schema, table1, database, null, "T%")));
+  }
+
+  @Test
+  public void testPatternSearchAllowedForPrimaryAndForeignKeys() throws SQLException {
+    Map<String, String> params = getConnectionParameters();
+    Properties properties = new Properties();
+    for (Map.Entry<?, ?> entry : params.entrySet()) {
+      if (entry.getValue() != null) {
+        properties.put(entry.getKey(), entry.getValue());
+      }
+    }
+    // test out connection parameter stringsQuoted to remove strings from quotes
+    properties.put("enablePatternSearch", "true");
+    Connection connection = DriverManager.getConnection(params.get("uri"), properties);
+
+    String database = connection.getCatalog();
+    String schema = connection.getSchema();
+    String schemaPattern = schema.substring(0, 1).concat("%");
+    final String table1 = "T1";
+    final String table2 = "T2";
+
+    connection
+        .createStatement()
+        .execute("create or replace table " + table1 + "(C1 int primary key, C2 string)");
+    connection
+        .createStatement()
+        .execute(
+            "create or replace table "
+                + table2
+                + "(C1 int primary key, C2 string, C3 int references "
+                + table1
+                + ")");
+
+    DatabaseMetaData dbmd = connection.getMetaData();
+
+    // Should return result for matching schema and table name
+    assertEquals(1, getSizeOfResultSet(dbmd.getPrimaryKeys(database, schema, table1)));
+
+    // Should return a result if we try a pattern match on the schema
+    assertTrue(0 < getSizeOfResultSet(dbmd.getPrimaryKeys(database, schemaPattern, table1)));
```

Comment:
let's use `assertThat` + `greaterThan` instead of `assertTrue` - the assert message is then more meaningful